pkg/test/spacebindingrequest/spacebindingrequest_assertions.go

```diff
@@ -1,10 +1,98 @@
 package spacebindingrequest
 
 import (
+	"context"
+
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type Assertion struct {
+	spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest
+	client              runtimeclient.Client
+	namespacedName      types.NamespacedName
+	t                   test.T
+}
+
+func (a *Assertion) loadResource() error {
+	spaceBindingRequest := &toolchainv1alpha1.SpaceBindingRequest{}
+	err := a.client.Get(context.TODO(), a.namespacedName, spaceBindingRequest)
+	a.spaceBindingRequest = spaceBindingRequest
+	return err
+}
+
+// AssertThatSpaceBindingRequest helper func to begin with the assertions on a SpaceBindingRequests
+func AssertThatSpaceBindingRequest(t test.T, namespace, name string, client runtimeclient.Client) *Assertion {
+	return &Assertion{
+		client:         client,
+		namespacedName: test.NamespacedName(namespace, name),
+		t:              t,
+	}
+}
+
+func (a *Assertion) Exists() *Assertion {
+	err := a.loadResource()
+	require.NoError(a.t, err)
+	return a
+}
+
+func (a *Assertion) HasFinalizer() *Assertion {
```

Comment:
Could we think of renaming this function to maybe HasToolChainFinalizer or HasSandboxFinalizer since this function is looking for a particular finalizer?
Or can make this generic and pass the name of the finalizer as argument

Reply:
I agree that `HasToolChainFinalizer` would be a better name. I would keep this name just for consistency , since all the other test utility functions in this repo have the same function name `HasFinalizer` see: https://github.com/search?q=repo%3Acodeready-toolchain%2Ftoolchain-common%20HasFinalizer&type=code

I would rather rename them all later and maybe introduce generics to have a single function for all those test packages (if possible) ? 

WDYT?